app/models/answer.rb

```diff
@@ -2,4 +2,11 @@ class Answer < ApplicationRecord
   belongs_to :question
 
   scope :correct, -> { where(correct: true) }
+
+  validate :answers_per_question, on: :create
+
+  def answers_per_question
+    answers_num = Question.find(question_id).answers.size
```

Comment:
```suggestion
    answers_num = question.answers.count
```
1. используй ассоциацию
2. я бы тебе рекомендовал использовать здесь `count` и проверку `count >= 4`, потому что тебя интересуют именно хранящиеся в базе ответы, точнее их количество
`size` же при наличии объектов в памяти берет их из памяти
в памяти же может быть уже "перебор" и объект не создастся, при этом в базе еще "недобор"

по поводу методов подсчета объектов `count`, `size`, `length` - посмотри для самообразования [вот этот вопрос на СО](https://stackoverflow.com/questions/6083219/activerecord-size-vs-count) и конечно же ответы к нему
1-ый и 2-ой ответы довольно лаконично и понятно объясняют что делают методы